src/hotspot/cpu/aarch64/stubGenerator_aarch64.cpp

```diff
@@ -5571,6 +5573,82 @@ class StubGenerator: public StubCodeGenerator {
     return start;
   }
 
+  // ARMv8.1 LSE versions of the atomic stubs used by Atomic::PlatformXX.
+  //
+  // If LSE is in use, generate LSE versions of all the stubs. The
+  // non-LSE versions are in atomic_aarch64.S.
```

Comment:
IMO it would be better for maintainability if the LSE versions were in atomic_aarch64.S too (with an explicit `.arch armv8-a+lse` directive). Is there any reason to generate them here, other than to support old toolchains? As far as I can tell GNU as supported LSE as far back as binutils 2.27.

https://sourceware.org/binutils/docs-2.27/as/AArch64-Extensions.html

Reply:
I can't see any reason to do this.There's be no benefit to moving this stuff, and it would be harder to change in the future. I'd do the whole lot as runtime stubs if I could, but they're needed before VM startup.